src/tools/r2rdump/R2RReader.cs

```diff
@@ -77,13 +83,65 @@ public unsafe R2RReader(string filename)
                 }
 
                 DirectoryEntry r2rHeaderDirectory = peReader.PEHeaders.CorHeader.ManagedNativeHeaderDirectory;
-                int r2rHeaderOffset = r2rHeaderDirectory.RelativeVirtualAddress - textSection.VirtualAddress + textSection.PointerToRawData;
-                R2RHeader = new R2RHeader(_image, (uint)r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
+                int r2rHeaderOffset = GetOffset(r2rHeaderDirectory.RelativeVirtualAddress, textSection);
+                R2RHeader = new R2RHeader(_image, r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
                 if (r2rHeaderDirectory.Size != R2RHeader.Size)
                 {
                     throw new System.BadImageFormatException("The calculated size of the R2RHeader doesn't match the size saved in the ManagedNativeHeaderDirectory");
                 }
+
+                R2RSection runtimeFunctions = R2RHeader.Sections[R2RSection.SectionType.READYTORUN_SECTION_RUNTIME_FUNCTIONS];
+                int curOffset = GetOffset((int)runtimeFunctions.RelativeVirtualAddress, textSection);
+                int nNativeCode = runtimeFunctions.Size / (3*sizeof(int));
+                NativeCode = new NativeCode[nNativeCode];
+                for (int i=0; i<nNativeCode; i++)
+                {
+                    int nativeCodeStartRva = GetInt32(_image, ref curOffset);
+                    int nativeCodeEndRva = GetInt32(_image, ref curOffset);
+                    int nativeCodeUnwindRva = GetInt32(_image, ref curOffset);
+                    NativeCode[i] = new NativeCode(_image, nativeCodeStartRva, nativeCodeEndRva, nativeCodeUnwindRva);
+                }
             }
         }
+
+        public int GetOffset(int rva, SectionHeader textSection)
```

Comment:
@jkotas would it be too confusing to make this an extension method so it appears that SectionHeader has an instance method of GetOffset(int rva)?

Reply:
Ok, that makes sense. Amy, you'll want to implement a search similar to what's done intenerally in PEHeaders.cs: https://github.com/dotnet/corefx/blob/f9573c604222db87fa74372fecc3891262fe0841/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaders.cs#L323

Specifically, you can use the public method [GetContainingSectionIndex](https://github.com/dotnet/corefx/blob/f9573c604222db87fa74372fecc3891262fe0841/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaders.cs#L350) to help you out